README.md
# Profiles REST API

Profiles REST API course code.

This is a User profile rest api written in django rest model framework.
Once deployed, the admin of the api must provide user with account username 
and password to authenticate and send different http requesto to interact with the API.



Requirement: chmod +x deploy/*.sh


deploy/history.txt
    1  ls
    2  python
    3  ls
    4  cd /
    5  ls
    6  cd home
    7  ls
    8  cd vagrant/
    9  ls
   10  touch test.txt
   11  ls
   12  rm -rf test.txt 
   13  ls
   14  cd /
   15  ls
   16  whereis python
   17  exit
   18  ls
   19  cd /
   20  ls
   21  cd profiles-rest-api/
   22  ls
   23  python hello.py 
   24  ls
   25  clear
   26  ls
   27  xs
   28  cd
   29  pwd
   30  clear
   31  python -m venv ~/env
   32  source ~/env/bin/activate
   33  deactivate
   34  source ~/env/bin/activate
   35  ls
   36  cd 
   37  ls
   38  cd /
   39  ls
   40  cd profiles-rest-api/
   41  ls
   42  cd
   43  ls
   44  pip install -r /profiles-rest-api/requirements.txt 
   45  ls
   46  cd /profiles-rest-api/
   47  ls;
   48  ls
   49  django-admin.py startproject profiles_project .
   50  ls
   51  python manage.py startapp profiles_api
   52  python manage.py runserver 0.0.0.0:8000
   53  ls
   54  pip3 install -r requirements.txt 
   55  pip install -r requirements.txt 
   56  pip install djangorestframework==3.9.2
   57  python manage.py runserver 0.0.0.0:8000
   58  cd
   59  ls
   60  python manage.py runserver 0.0.0.0:8000
   61  python /profiles-rest-api/manage.py runserver 0.0.0.0:8000
   62  ls
   63  cd 
   64  l;s
   65  ls
   66  cd /
   67  ls
   68  cd profiles-rest-api/
   69  ls
   70  rm -rf manage.py profiles_api profiles_project
   71  ls
   72  django-admin.py startproject profiles_project .
   73  ls
   74  python manage.py startapp profiles_api
   75  ls
   76  cd
   77  python /profiles-rest-api/manage.py runserver 0.0.0.0:8000
   78  ls
   79  source ~/env/bin/activate
   80  ls
   81  python /profiles-rest-api/manage.py makemigrations profiles_api
   82  ls
   83  cd/
   84  ls
   85  cd /
   86  ls
   87  cd profiles-rest-api/
   88  ls
   89  python manage.py makemigrations profiles_api
   90  cd profiles-rest-api/
   91  python manage.py makemigrations profiles_api
   92  exit
   93  ls
   94  source ~/env/bin/activate
   95  ls
   96  python /profiles-rest-api/manage.py makemigrations profiles_api
   97  ls
   98  python /profiles-rest-api/manage.py migrate
   99  ls
  100  cd /profiles-rest-api/
  101  ls
  102  python manage.py createsuperuser
  103  python manage.py runserver 0.0.0.0:8000
  104  python manage.py createsuperuser
  105  python manage.py runserver 0.0.0.0:8000
  106  ls
  107  cd profiles_
  108  ls
  109  cd profiles_
  110  cd profiles_api/
  111  ls
  112  touch urls.py
  113  python manage.py runserver 0.0.0.0:8000
  114  ls
  115  cd ..
  116  ls
  117  python manage.py runserver 0.0.0.0:8000
  118  ls
  119  cd profiles_
  120  ls
  121  cd profiles_api/
  122  ls
  123  touch serializer.py
  124  python manage.py runserver 0.0.0.0:8000
  125  LS
  126  ls
  127  cd .
  128  ls
  129  cd ..
  130  ls
  131  python manage.py runserver 0.0.0.0:8000
  132  python manage.py runserver --noreload 0.0.0.0:8000
  133  exit
  134  ls
  135  source ~/env/bin/activate
  136  ls
  137  history
  138  ls
  139  cd /
  140  ls
  141  clear
  142  ls
  143  cd profiles-rest-api/
  144  ls
  145  cd profiles-rest-api/
  146  python manage.py runserver --noreload 0.0.0.0:8000
  147  python manage.py runserver  0.0.0.0:8000
  148  source ~/env/bin/activate
  149  cd /
  150  ls
  151  cd profiles-rest-api/
  152  c
  153  clear
  154  ls
  155  python manage.py runserver 0.0.0.0:8000
  156  clear
  157  source ~/env/bin/activate
  158  ls
  159  python manage.py runserver 0.0.0.0:8000
  160  cd
  161  exirt
  162  exitr
  163  exit
  164  source ~/env/bin/activate
  165  ls
  166  cd /profiles-rest-api/
  167  python manage.py runserver 0.0.0.0:8000
  168  git status
  169  python manage.py runserver 0.0.0.0:8000
  170  ls
  171  cd profiles_
  172  cd profiles_api
  173  python manage.py runserver 0.0.0.0:8000
  174  cd ..
  175  python manage.py runserver 0.0.0.0:8000
  176  python manage.py runserver --noreload 0.0.0.0:8000
  177  python manage.py runserver 0.0.0.0:8000
  178  python manage.py runserver --noreload 0.0.0.0:8000
  179  exit
  180  cd /profiles-rest-api/
  181  ls
  182  clear
  183  python manage.py runserver --noreload 0.0.0.0:8000
  184  source ~/env/bin/activate
  185  python manage.py runserver --noreload 0.0.0.0:8000
  186  python manage.py runserver 0.0.0.0:8000
  187  exit
  188  ls
  189  source ~/env/bin/activate
  190  ls
  191  cd /profiles-rest-api/
  192  ls
  193  python manage.py runserver 0.0.0.0:8000
  194  cd profiles_
  195  python manage.py runserver 0.0.0.0:8000
  196  python manage makemigration
  197  ls
  198  ls -la
  199  python manage.py makemigrations
  200  python manage.py migratew
  201  python manage.py migrate
  202  ls
  203  python manage.py runserver 0.0.0.0:8000
  204  python manage.py collectstatic --noinput
  205  export PROJECT_BASE_PATH='/usr/local/apps/profiles-rest-api'
  206  env
  207  history
  208  history >> history.txt


profiles_api/migrations/0001_initial.py
# Generated by Django 2.2 on 2020-08-24 20:55

from django.db import migrations, models


class Migration(migrations.Migration):

    initial = True

    dependencies = [
        ('auth', '0011_update_proxy_permissions'),
    ]

    operations = [
        migrations.CreateModel(
            name='UserProfile',
            fields=[
                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('password', models.CharField(max_length=128, verbose_name='password')),
                ('last_login', models.DateTimeField(blank=True, null=True, verbose_name='last login')),
                ('is_superuser', models.BooleanField(default=False, help_text='Designates that this user has all permissions without explicitly assigning them.', verbose_name='superuser status')),
                ('email', models.EmailField(max_length=255, unique=True)),
                ('name', models.CharField(max_length=255)),
                ('is_active', models.BooleanField(default=True)),
                ('is_staff', models.BooleanField(default=False)),
                ('groups', models.ManyToManyField(blank=True, help_text='The groups this user belongs to. A user will get all permissions granted to each of their groups.', related_name='user_set', related_query_name='user', to='auth.Group', verbose_name='groups')),
                ('user_permissions', models.ManyToManyField(blank=True, help_text='Specific permissions for this user.', related_name='user_set', related_query_name='user', to='auth.Permission', verbose_name='user permissions')),
            ],
            options={
                'abstract': False,
            },
        ),
    ]
